File: experiments/runs/2026-09-20_07-22-29_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/Claims.java
import java.math.BigDecimal;
import java.math.RoundingMode;

/** Computes payouts for damage reported against a policy. */
final class Claims {

    private static final BigDecimal DEDUCTIBLE = BigDecimal.valueOf(100);
    private static final BigDecimal HALF = new BigDecimal("0.5");
    private static final int HALF_REIMBURSEMENT_ENCHANTMENT = 8;

    private Claims() {
    }

    /**
     * The reimbursement for one damaged item, after its clause and the deductible.
     *
     * <p>Dragon material is fully reimbursed, which is also the default treatment, so it
     * needs no branch of its own; the 50 % clause for highly enchanted items wins over it.
     */
    static BigDecimal itemPayout(Item item, BigDecimal damage) {
        BigDecimal reimbursed = damage;
        if (item.enchantment() >= HALF_REIMBURSEMENT_ENCHANTMENT) {
            reimbursed = reimbursed.multiply(HALF);
        }
        BigDecimal afterDeductible = reimbursed.subtract(DEDUCTIBLE);
        return afterDeductible.max(BigDecimal.ZERO);
    }

    /** Payouts are rounded down — in the MHPCO's favour. */
    static int roundPayout(BigDecimal amount) {
        return amount.setScale(0, RoundingMode.FLOOR).intValueExact();
    }
}

File: experiments/runs/2026-09-22_00-17-13_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/claims.py
"""Claim processing for MHPCO policies."""

from fractions import Fraction

DEDUCTIBLE = 100
HIGH_ENCHANTMENT_CLAIM_LEVEL = 8
HIGH_ENCHANTMENT_SHARE = Fraction(1, 2)
DRAGON_MATERIAL = "dragon"


def round_down(amount):
    """Payouts round in the MHPCO's favour."""
    return Fraction(amount) // 1


def damage_payout(item, amount):
    """Reimbursement for a single damaged item, after its own deductible.

    The high-enchantment clause takes precedence over dragon material when
    both apply; the deductible is taken last.
    """
    reimbursed = Fraction(amount)
    if item.get("enchantment", 0) >= HIGH_ENCHANTMENT_CLAIM_LEVEL:
        reimbursed *= HIGH_ENCHANTMENT_SHARE
    return max(Fraction(0), reimbursed - DEDUCTIBLE)
